packages/splitter/src/useSplitter.ts

```diff
@@ -53,46 +72,73 @@ export interface IUseSplitterProps extends Omit<HTMLProps<any>, 'onChange'> {
   valueNow?: number;
   /** A callback that receives updated valueNow when a separator moves */
   onChange?: (valueNow: number) => void;
+  /** An enum that specifies whether the separator leads or trails the primary pane */
+  position: SplitterPosition;
   /** A boolean that changes splitter from left-to-right to right-to-left mode */
   rtl?: boolean;
 }
 
 export interface IUseSplitterReturnValue {
   getSeparatorProps: <T>(options?: T & HTMLProps<any>) => any;
+  getPrimaryPaneProps: <T>(options?: T & HTMLProps<any>) => any;
+  valueNow: number;
 }
 
-// Pointer coordinates do not account padding, margins, or height/width of separator
-// This function takes mouse or touch value minus padding or margin position minus the width or height of the separator divided by 2
-// This aligns the pointer and the separator correctly
-// When passed a viewportWidth, subtract pointer position and offsets to invert the calculation for RTL mode.
-export const calculateOffset = (
+// takes pointer distance from edge of viewport to the pointer
+// minus the offset of pointer position to the separator in the DOM layout
+// minus the separator width or height divided by 2 to place cursor in middle of separator
+export const normalizePointerToSeparator = (
+  bodyPadding: number,
   pointerPosition: number,
-  paddingOrMarginPosition = 0,
-  offsetDimension = 0,
-  viewportWidth = 0
+  separatorHeightOrWidth = 0,
+  viewportWidthOrHeight = 0
 ) => {
-  if (viewportWidth === 0) {
-    return pointerPosition - paddingOrMarginPosition - Math.ceil(offsetDimension / 2);
+  // if placement trails use the pointerPosition relative to the left of the viewport
+  if (viewportWidthOrHeight === 0) {
+    return pointerPosition - bodyPadding - Math.floor(separatorHeightOrWidth / 2);
   }
 
-  return viewportWidth - pointerPosition - paddingOrMarginPosition - offsetDimension;
+  // if position leads we need to invert the pointerPosition so its relative to right of the viewport
+  return (
+    viewportWidthOrHeight - pointerPosition - bodyPadding - Math.floor(separatorHeightOrWidth / 2)
+  );
 };
 
 export const verticalArrowKeys = {
   rtl: {
-    INCREASE: 'ArrowLeft',
-    DECREASE: 'ArrowRight'
+    [SplitterPosition.TRAILS]: {
+      INCREASE: 'ArrowLeft',
+      DECREASE: 'ArrowRight'
+    },
+    [SplitterPosition.LEADS]: {
+      INCREASE: 'ArrowRight',
+      DECREASE: 'ArrowLeft'
+    }
   },
   ltr: {
-    INCREASE: 'ArrowRight',
-    DECREASE: 'ArrowLeft'
+    [SplitterPosition.LEADS]: {
+      INCREASE: 'ArrowLeft',
+      DECREASE: 'ArrowRight'
+    },
+    [SplitterPosition.TRAILS]: {
+      INCREASE: 'ArrowRight',
+      DECREASE: 'ArrowLeft'
+    }
   }
 };
 
+const xor = (a: boolean | undefined, b: boolean | undefined) => {
+  if (a && b) {
+    return false;
+  }
+
+  return a || b;
+};
+
 export function useSplitter({
+  idPrefix,
   ariaLabel,
-  primaryPaneId,
-  environment = document,
+  windowObject = window,
```

Comment:
This change was made in light of https://github.com/zendeskgarden/react-containers/pull/405#discussion_r767817695

We should revisit this convention in `v9` and I have added this to the agenda accordingly.

Reply:
I wonder if a complementary `useWindow` hook safely checks/extracts the value would be one way of handling this.

Reply:
👍 re: `useWindow` as a compliment to `useDocument`. And we're not necessarily gated here by `react-components` v9 since this repo uses independent versioning.

Reply:
Edit: `environment` is `document` not `window` which I incorrectly mentioned above.

Source: https://github.com/zendeskgarden/react-containers/blob/main/packages/focusjail/src/useFocusJail.ts#L19

Reply:
Added useWindow and migrated back to environment per discussion. https://github.com/zendeskgarden/react-containers/pull/405/commits/04b21508933cf7ae55b3b61e795cb2c365789196